core/src/main/java/com/graphhopper/routing/util/parsers/BikeCommonAccessParser.java

```diff
@@ -42,61 +37,87 @@ protected BikeCommonAccessParser(BooleanEncodedValue accessEnc, BooleanEncodedVa
-                if (bikeTag == null && !way.hasTag("foot") || intendedValues.contains(bikeTag))
-                    access = WayAccess.FERRY;
-            }
+        if (!allowedHighways.contains(highwayValue))
+            return WayAccess.CAN_SKIP;
+
+        String bicycleValue = way.getTag("bicycle");
+        boolean hasAllowedBikesExplicitly = intendedValues.contains(bicycleValue);
 
-            // special case not for all acceptedRailways, only platform
-            if (way.hasTag("railway", "platform"))
-                access = WayAccess.WAY;
+        // use the way if it is tagged for bikes, even if for dismount (speed-wise: PUSHING_SECTION_SPEED)
+        if ("dismount".equals(bicycleValue))
+            return WayAccess.WAY;
 
-            if (way.hasTag("man_made", "pier"))
-                access = WayAccess.WAY;
+        // check tricky ways, that DO NOT have explicit tags like bicycle=yes
+        if (!hasAllowedBikesExplicitly) {
+            if ("motorway".equals(highwayValue) || "motorway_link".equals(highwayValue))
+                return WayAccess.CAN_SKIP;
 
-            if (!access.canSkip()) {
-                if (way.hasTag(restrictionKeys, restrictedValues))
-                    return WayAccess.CAN_SKIP;
-                return access;
-            }
+            if (way.hasTag("motorroad", "yes"))
+                return WayAccess.CAN_SKIP;
 
-            return WayAccess.CAN_SKIP;
+            if (isBlockFords() && ("ford".equals(highwayValue) || way.hasTag("ford")))
+                return WayAccess.CAN_SKIP;
         }
 
-        if (!allowedHighways.contains(highwayValue))
-            return WayAccess.CAN_SKIP;
+        // check explicit access restrictions
+        WayAccess explicitRestriction = checkExplicitRestrictionsHierarchy(way);
+        if (explicitRestriction != WayAccess.OTHER) {
+            return explicitRestriction;
+        }
 
-        // use the way for pushing
-        if (way.hasTag("bicycle", "dismount"))
+        return WayAccess.WAY;
+    }
+
+    private WayAccess checkNonHighwayAccess(ReaderWay way) {
+        // special case not for all acceptedRailways, only platform
+        if (way.hasTag("railway", "platform"))
             return WayAccess.WAY;
 
-        int firstIndex = way.getFirstIndex(restrictionKeys);
-        if (firstIndex >= 0) {
-            String firstValue = way.getTag(restrictionKeys.get(firstIndex), "");
-            String[] restrict = firstValue.split(";");
-            for (String value : restrict) {
-                if (restrictedValues.contains(value) && !hasTemporalRestriction(way, firstIndex, restrictionKeys))
-                    return WayAccess.CAN_SKIP;
-                if (intendedValues.contains(value))
-                    return WayAccess.WAY;
-            }
+        if (way.hasTag("man_made", "pier"))
+            return WayAccess.WAY;
+
+
+        WayAccess explicitRestriction = checkExplicitRestrictionsHierarchy(way);
```

Comment:
This is a breaking change, no? So every ferry that has not set bike=no is suddenly allowed? I am not sure this is necessarily true.

I would furthermore argue, that if we change this for bike, this should also change for the other access parsers?

All that said, GraphHopper is usually rather strict than loose, so the idea is to block stuff where we are not sure.

Reply:
At very least this needs to be documented in the changelog. But I am not sure this will go through.

Reply:
Yes, this is definitely up to debate, ideally this would be something to configure? is there a precedent for something like this?

I can change it to "block where we are not sure" and update the tests, but still, I like the direction of this refactor to have explicit value and then act upon it.

Reply:
There is for example the blockPrivate, I think this could be something that is implemented like this?

Reply:
yup, but then they want to remove block_private too :D maybe another way is needed

https://github.com/graphhopper/graphhopper/pull/3076